### ISSUE TITLE
Offer camelCase suggestions for  spelling corrections

### ISSUE THREAD

**Issue by janice143** [streetsidesoftware/vscode-spell-checker#2882]
this plugin provides the ability to check camelCase words by splitting it into a list of known English words, such as

camelCase -> camel case
HTMLInput -> html input – Notice that the I is associated with Input and not HTML
snake_case_words -> snake case words
camel2snake -> camel snake – (the 2 is ignored)

This means any two or more words can be put together to make a camelCase new word which should become a suggestion spelling in the suggestion list.

so, can you add a feature to support camelCase word suggestion for speling correstions?

**Comment by Jason3S** [comment]
@janice143,

Would you give some more examples? I'm trying to understand exactly what you are looking for.

Are you hoping that `camelcase` will result the in the suggestion of `camelCase`? Or do you already have a camel case word in your dictionary and want it to be suggested?

**Comment by janice143** [comment]
Yes, I'm hoping camelcase results will be suggestions for correction without manually adding them in the dictionary. Most of time when I'm coding, I tend to make this kind of spelling mistake that the word is supposed to be in cammel case that can pass your plugin. Such as,




`setloading` in the following code snippet, which should be correct as `setLoading`


const [loading, setloading] = useState();




or a common variable `datasource`, which should be fix as `dataSource`.




It would be great if the correct camel case style could be included in the list of fixing suggestions. 




I read your document about how the plugin handle the camelCase ( shown as the following pic), I thougth since you can identify misspelled camel case words forwardly, is it possible to add a feature to reversely correct words to the camel case spelling.











                       
Original Email
                       
                     

Sender:"Jason Dent"< ***@***.*** &gt;;

Sent Time:2023/10/22 20:05

To:"streetsidesoftware/vscode-spell-checker"< ***@***.*** &gt;;

Cc recipient:"rubber duck"< ***@***.*** &gt;;"Mention"< ***@***.*** &gt;;

Subject:Re: [streetsidesoftware/vscode-spell-checker] Offer camelCasesuggestions for  spelling corrections (Issue #2882)




 
@janice143,
 
Would you give some more examples? I'm trying to understand exactly what you are looking for.
 
Are you hoping that camelcase will result the in the suggestion of camelCase? Or do you already have a camel case word in your dictionary and want it to be suggested?
